### ISSUE TITLE
Please don't ignore the supplied CXXFLAGS

### ISSUE THREAD

**Issue by yurivict** [octobanana/peaclock#13]
This change is needed:
```
--- CMakeLists.txt.orig 2019-05-23 00:24:50 UTC
+++ CMakeLists.txt
@@ -13,7 +13,7 @@ endif (CCACHE_FOUND)
 set (CMAKE_CXX_STANDARD 17)
 set (CMAKE_CXX_STANDARD_REQUIRED ON)
 
-set (CMAKE_CXX_FLAGS "-std=c++17 -fdiagnostics-color=auto")
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -fdiagnostics-color=auto")
 
 set (DEBUG_FLAGS "-g -Og -rdynamic -Wpedantic -Wall -Wextra -Wcast-align -Wcast-qual -Wctor-dtor-privacy -Wdisabled-optimization -Wformat=2 -Winit-self -Wlogical-op -Wmissing-declarations -Wmissing-include-dirs -Wnoexcept -Wold-style-cast -Woverloaded-virtual -Wredundant-decls -Wshadow -Wsign-conversion -Wsign-promo -Wstrict-null-sentinel -Wstrict-overflow=5 -Wswitch-default -Wundef -Wno-unused")
 set (DEBUG_LINK_FLAGS "-fprofile-arcs -ftest-coverage")
```

**Comment by octobanana** [comment]
>CXXFLAGS
>
>This is a CMake Environment Variable. Its initial value is taken from the calling process environment.
>
>Default compilation flags to be used when compiling CXX (C++) files. Will only be used by CMake on the first configuration to determine CXX default compilation flags, after which the value for CXXFLAGS is stored in the cache as CMAKE_CXX_FLAGS. For any configuration run (including the first), the environment variable will be ignored if the CMAKE_CXX_FLAGS variable is defined.

So the following appends any additional flags to the variable without overwriting any default flags set by the CXXFLAGS environment variable.

```diff
-set (CMAKE_CXX_FLAGS "-std=c++17 -fdiagnostics-color=auto")
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++17 -fdiagnostics-color=auto")
```

Makes sense, thanks for the tip!

**Comment by octobanana** [comment]
Fixed in version [0.4.1](https://github.com/octobanana/peaclock/releases/tag/0.4.1), thanks!
